### ISSUE TITLE
[Enhancement]: extend addItemToCart for more complex product types

### ISSUE THREAD

**Issue by helgatheviking** [woothemes/woocommerce#53357]
### Describe the solution you'd like

The [`addItemtoCart()` thunk](https://github.com/woocommerce/woocommerce/blob/ef5a61cec9d11d6466599096d517bff3951938bc/plugins/woocommerce-blocks/assets/js/data/cart/thunks.ts#L199C14-L199C27) only supports product ID and quantity. This makes it impossible to use for asyncronously addinging configurable single products to the cart (such as even variable products, or simple products with product addons, - or in my case - products with a name your price field).

I would like to see it support sending additional parameters. The product collection [add to cart button uses the following code](https://github.com/woocommerce/woocommerce/blob/ef5a61cec9d11d6466599096d517bff3951938bc/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/button/frontend.tsx#L140-L143)

```
yield dispatch( storeKey ).addItemToCart(
					productId,
					quantityToAdd
				);
```

I would like to be able to do something similar a single product block (for now just as an extension, but ideally I think you should support that in core when the single add to cart form block is fully blockified)

### Describe alternatives you've considered

There's currently no alternative I can see that uses the StoreAPI or the cart redux data store. The only remaining options are to 1 use the ajax add to cart endpoint (honestly this probably will have much better cross-compatibility) or 2. rewrite a duplicate of `addItemtoCart` as my own custom thunk. 

### Additional context

At the moment I am exploring displaying a single product block and a checkout block on the same page. If I can add the item to the cart via the cart data store then I already have the checkout block to automatically show/hide depending on if the cart has items in it or not.

**Comment by senadir** [comment]
I agree, I don't see a reason why we wouldn't do this so adding this to backlog, feel free to push the change yourself if you want.

**Comment by helgatheviking** [comment]
@senadir I have proposed https://github.com/woocommerce/woocommerce/pull/53458
